system/CLI/CLI.php

```diff
@@ -225,12 +225,12 @@ public static function prompt(string $field, $options = null, $validation = null
         $extraOutput = '';
         $default     = '';
 
-        if ($validation && ! is_array($validation) && ! is_string($validation)) {
+        if (isset($validation) && ! is_array($validation) && ! is_string($validation)) {
             throw new InvalidArgumentException('$rules can only be of type string|array');
         }
 
         if (! is_array($validation)) {
-            $validation = $validation ? explode('|', $validation) : [];
+            $validation = ((string) $validation !== '') ? explode('|', $validation) : [];
```

Comment:
Skip the `explode()` function with an empty string

Reply:
That's true, but how many times do we set validation rules without specifying the rules? An empty string will be handled correctly with `explode()`.

Using typing makes the code less clear. I would like to avoid it if we can.

Reply:
Almost the same answer https://github.com/codeigniter4/CodeIgniter4/pull/9302#discussion_r1898667545

Reply:
An empty string as an edge case, I would not worry about it that much. For me, the readability of the code is more important.